gnucash/report/report-system/commodity-utilities.scm

```diff
@@ -23,49 +23,35 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; Functions to get splits with interesting data from accounts.
 
+;; helper function. queries book for all splits in accounts before
+;; end-date (end-date can be #f)
+(define (get-all-splits accounts end-date)
+  (let ((query (qof-query-create-for-splits)))
+    (qof-query-set-book query (gnc-get-current-book))
+    (gnc:query-set-match-non-voids-only! query (gnc-get-current-book))
+    (xaccQueryAddAccountMatch query accounts QOF-GUID-MATCH-ANY QOF-QUERY-AND)
+    (xaccQueryAddDateMatchTT query #f 0 (and end-date #t) (or end-date 0) QOF-QUERY-AND)
+    (let ((splits (qof-query-run query)))
+      (qof-query-destroy query)
+      splits)))
 
 ;; Returns a list of all splits in the 'currency-accounts' up to
 ;; 'end-date' which have two different commodities involved, one of
 ;; which is equivalent to 'commodity' (the latter constraint only if
 ;; 'commodity' != #f ).
 (define (gnc:get-match-commodity-splits
          currency-accounts end-date commodity)
-  (let ((query (qof-query-create-for-splits))
-        (splits #f))
-
-    (qof-query-set-book query (gnc-get-current-book))
-    (gnc:query-set-match-non-voids-only! query (gnc-get-current-book))
-    (xaccQueryAddAccountMatch query
-                              currency-accounts
-                              QOF-GUID-MATCH-ANY QOF-QUERY-AND)
-    (if end-date
-        (xaccQueryAddDateMatchTT
-         query #f end-date #t end-date QOF-QUERY-AND))
-
-    ;; Get the query result, i.e. all splits in currency
-    ;; accounts.
-    (set! splits (filter
-                  ;; Filter such that we get only those splits
-                  ;; which have two *different* commodities
-                  ;; involved.
-                  (lambda (s) (let ((trans-comm
-                                     (xaccTransGetCurrency
-                                      (xaccSplitGetParent s)))
-                                    (acc-comm
-                                     (xaccAccountGetCommodity
-                                      (xaccSplitGetAccount s))))
-                                (and
-                                 (not (gnc-commodity-equiv
-                                       trans-comm acc-comm))
-                                 (or
-                                  (not commodity)
-                                  (gnc-commodity-equiv
-                                   commodity trans-comm)
-                                  (gnc-commodity-equiv
-                                   commodity acc-comm)))))
-                  (qof-query-run query)))
-    (qof-query-destroy query)
-    splits))
+  ;; Filter such that we get only those splits which have two
+  ;; *different* commodities involved.
```

Comment:
As @jralls also mentioned in PR #414 this comment is incomplete. I had noticed this but eventually forgot to raise a comment here. As it is explained well in the whole function's comment, perhaps this one can be dropped ?